### DISCUSSION TITLE
Confusion around create_study and load_study with respect to pruner and sampler

### DISCUSSION THREAD

**Discussion by Guillaume227** [optuna/optuna#4883] [Q&A]
I am running a distributed study (using journal file storage) created with `create_study(study_name='my_study', storage='...')`. In each of the workers I use `load_study(study_name='my_study', storage='...')` to get the study instance before calling optimize.

When passing pruner and sampler arguments to `create_study` I was not getting the right result (e.g. no pruning when using NopPruner because each worker defaults to MedianPruner after calling `load_study`) until I realized `load_study` also expects those arguments.

What's the right way to think about this? Why should those arguments be repeated as opposed to being persisted with the study? Aren't pruner and sampler intrinsically part of a study definition and behavior?

Or is the way I am calling `create_study` / `load_study` not the right pattern? Looking at the documentation did not help clearing my confusion.

**Comment by nzw0301** [comment]
The storage does not keep a sampler and pruner instances. We need to create them in the same way amongst distributed processes.

https://optuna.readthedocs.io/en/stable/tutorial/20_recipes/001_rdb.html#resume-study
